Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -313,17 +313,28 @@ void ViewTransition::setupViewTransition()
     });
 }
 
-static AtomString effectiveViewTransitionName(RenderLayerModelObject& renderer, Element& originatingElement, Style::Scope& documentScope)
+static AtomString effectiveViewTransitionName(RenderLayerModelObject& renderer, Element& originatingElement, Style::Scope& documentScope, bool isCrossDocument)
 {
     if (renderer.isSkippedContent())
         return nullAtom();
     auto transitionName = renderer.style().viewTransitionName();
-    if (!transitionName)
+    if (transitionName.isNone())
         return nullAtom();
-    auto scope = Style::Scope::forOrdinal(originatingElement, transitionName->scopeOrdinal);
+    auto scope = Style::Scope::forOrdinal(originatingElement, transitionName.scopeOrdinal());
     if (!scope || scope != &documentScope)
         return nullAtom();
-    return transitionName->name;
+    if (transitionName.isCustomIdent())
+        return transitionName.customIdent();
+    ASSERT(transitionName.isAuto());
+    if (originatingElement.hasID())
+        return originatingElement.getIdAttribute();
```

Comment:
`originatingElement` actually isn't right in the context of pseudo-elements, you'll actually want `renderer.element()` here. In the case of pseudo-elements, you don't want `#foo::after` to transition with `#foo`.

Maybe worth a WPT and a comment.